lib/consensus.js

```diff
@@ -39,6 +41,36 @@ function isValidBlockHeader(newHeader, previousHeaders, network = 'mainnet') {
     && hasGreaterThanMedianTimestamp(newHeader, previousHeaders);
 }
 
+/**
+ * validates an array of transaction objects, an array of tx hashes
+ * or a single transaction object or its hash
+ * against a merkleblock and the local header chain
+ * @param {Object[]|string[]|Object|string} transactions
```

Comment:
Looks too complex. Is there reason to have such swiss knife? I would accept only an array of hashes.

Reply:
The only reason was because I didn't know exactly what we will need. If array of hashes is the only thing we need, it makes sense to offer only that.

Reply:
Do we need something else for validation except hashes? :))

Reply:
No :))). In either case it is just the tx hash which is validated. It would be just a convenience for the consumer of the function to be able to just pass the tx object without having to remember that he should use only tx.hash. I removed the ability to pass something other than array. But left the possibility to pass the object, as it is already handled in `MerkleBlock.prototype.hasTransaction`